### PR TITLE
enable pdf export and automatically upload to ipfs

### PR DESCRIPTION
**Pull request by emilianobonassi**
adding support to export to pdf, so it's easier to read offline (e.g. Kindle) and automatically update to ipfs

### CODE REVIEW

**Comment by smartcontracts** [comment]
Wow, this is awesome. Let me test this locally.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Build
+
+on:
+  push:
+    branches:
+    - main
+
+  pull_request:
```
Comment (smartcontracts) [inline .github/workflows/build.yaml:8]: I think it should be good enough to only publish to IPFS on push (might be a lot of content to push to IPFS if it's uploaded for every PR).

Comment (emilianobonassi) [reply .github/workflows/build.yaml:8]: agree, will fix

Comment (emilianobonassi) [reply .github/workflows/build.yaml:8]: fixed

**Comment by smartcontracts** [comment]
Ok sweet, just need to resolve the last merge conflicts and we can get this merged (might be easiest to simply remove `yarn.lock` and regenerate it). Thank you! This is sick.

**Comment by emilianobonassi** [comment]
yeah, currently on mobile, will resolve tomorrow

great job the book!

**Comment by smartcontracts** [comment]
> yeah, currently on mobile, will resolve tomorrow
> 
> great job the book!

No rush at all. And thank you, it's still a work in progress but starting to ramp back up again :-) Hit me with an ETH address and I'll send you a contributor NFT!

**Comment by emilianobonassi** [comment]
@smartcontracts conflicts managed, ready to go!

here my ENS: emiliano.eth

File: crowdin.yml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 files:
-  - source: /book/**/*.*
+  - source: book/**/*.md
```
Comment (smartcontracts) [inline crowdin.yml]: One final comment here, let's revert this change (since we need to include other files types besides markdown in the translations).

Comment (emilianobonassi) [reply crowdin.yml]: fixed

**Review by smartcontracts — APPROVED** [review]
LGTM, thank you!
